### PR TITLE
Added 'MartianColors' enum and colored the app.

### PR DESCRIPTION
**Pull request by matej494**

### CODE REVIEW

**Comment by dkulundzic** [comment]
screen shot 2018-07-18 at 14 27 16

The project is missing the `Utilities`, `Network` and `Persistence` folders. This usually happens when you create them in Xcode, but they're empty. `Git` does not track empty folders. 

To solve the issue, we can add a dummy file in each empty folder. A "convention" is to create a empty file and name it `.gitkeep`. Or, do not add empty folders within Xcode - add them only when you have content to fill these folders up 🙂 

Invoke the `touch .gitkeep`command within the empty folder (via the Terminal) to create this file.

Please resolve before proceeding.

File: Marsiple/Common/UI/MartianColors.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+//  MartianColors.swift
+//  Marsiple
+//
+//  Created by Matej Korman on 17/07/2018.
+//  Copyright © 2018 Matej Korman. All rights reserved.
+//
+
+import UIKit
+
+enum MatrianColor {
```
Comment (dkulundzic) [inline Marsiple/Common/UI/MartianColors.swift]: Typo in `enum` name - `MatrianColor` -> `MartianColor`.

File: Marsiple/Common/UI/MartianColors.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    case red
+    case dark
+    case darkGray
+    case gray
+    case lightGray
+    
+    var value: UIColor {
+        switch self {
+        case .red:
+            return UIColor(ciColor: CIColor(red: 0.937, green: 0.251, blue: 0.349))
```
Comment (dkulundzic) [inline Marsiple/Common/UI/MartianColors.swift]: We shouldn't use the `CIColor` to create colors - there's a more straightforward and simpler approach. 

Add a `UIColor` extension with a new convenience initialiser: 

```
public convenience init(red: Int, green: Int, blue: Int, alpha: CGFloat = 1) {
		self.init(red: CGFloat(red) / 255, green: CGFloat(green) / 255, blue: CGFloat(blue) / 255, alpha: alpha)
}
```

File: Marsiple/App/Tab Bar/TabBarController.swift [hunk 3]
```diff
@@ -13,6 +13,9 @@ class TabBarController: UITabBarController {
         super.init(nibName: nil, bundle: nil)
         let navigationController = UINavigationController(rootViewController: PostsViewController())
         navigationController.navigationBar.prefersLargeTitles = true
+        navigationController.navigationBar.barTintColor = MatrianColor.gray.value
```
Comment (dkulundzic) [inline Marsiple/App/Tab Bar/TabBarController.swift]: Wouldn't it be cool if we could use custom colors like-so:
`navigationController.navigationBar.barTintColor = .martianGray`

Do you have an idea how you would accomplish this?

**Review by dkulundzic — CHANGES_REQUESTED** [review]
Looks great! Left a couple of remarks.

File: Marsiple/App/Tab Bar/TabBarController.swift (hunk 3 above)
Comment (matej494) [reply Marsiple/App/Tab Bar/TabBarController.swift]: We could do this in `UIColor`extension using computed properties.

File: Marsiple/Common/Extensions/UIColor-Marsiple.swift [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+//  UIColor.swift
+//  Marsiple
+//
+//  Created by Matej Korman on 18/07/2018.
+//  Copyright © 2018 Matej Korman. All rights reserved.
+//
+
+import UIKit
+
+extension UIColor {
```
Comment (dkulundzic) [inline Marsiple/Common/Extensions/UIColor-Marsiple.swift:11]: Very good, nice job! 👍 

P.s. How do YOU like this approach?

File: Marsiple/App/Posts/PostTableViewCell.swift [hunk 5]
```diff
@@ -32,7 +32,7 @@ extension PostTableViewCell {
 private extension PostTableViewCell {
     func setupViews() {
         accessoryType = .disclosureIndicator
-        backgroundColor = MatrianColor.lightGray.value
+        backgroundColor = .martianLightGrey
```
Comment (dkulundzic) [inline Marsiple/App/Posts/PostTableViewCell.swift]: Much shorter and much better 👍

**Review by dkulundzic — CHANGES_REQUESTED** [review]
Looking great, well done - lets rock 🚀 

Edit: Whoops, accidentally clicked `Request changes` on the review.

**Review by dkulundzic — APPROVED** [review]

File: Marsiple/Common/Extensions/UIColor-Marsiple.swift (hunk 4 above)
Comment (matej494) [reply Marsiple/Common/Extensions/UIColor-Marsiple.swift:11]: It's better the the previous one. Very clean call site.

**Comment by matej494** [comment]
Successfully merged branch 'adding-color' into branch 'posts' without conflicts.

**Pull request merged**